applications/asset_tracker/src/main.c

```diff
@@ -631,98 +636,53 @@ void sensors_start(void)
 /**@brief nRF Cloud specific callback for cloud association event. */
 static void on_user_pairing_req(const struct cloud_event *evt)
 {
-	if (evt->data.pair_info.type == CLOUD_PAIR_SEQUENCE) {
-		if (!atomic_get(&pattern_recording)) {
-			ui_led_set_pattern(UI_CLOUD_PAIRING);
-			atomic_set(&pattern_recording, 1);
-			buttons_captured = 0;
-			buttons_to_capture = *evt->data.pair_info.buf;
-
-			printk("Please enter the user association pattern ");
-			printk("using the buttons and switches\n");
-		}
-	} else if (evt->data.pair_info.type == CLOUD_PAIR_PIN) {
-		association_with_pin = true;
+	if (atomic_get(&association_requested) == 0) {
+		atomic_set(&association_requested, 1);
 		ui_led_set_pattern(UI_CLOUD_PAIRING);
-		printk("Waiting for cloud association with PIN\n");
-	}
-}
+		printk("Add device to cloud account.\n");
+		printk("Waiting for cloud association...\n");
 
-#if defined(CONFIG_USE_UI_MODULE)
-/**@brief Send user association information to nRF Cloud. */
-static void cloud_user_associate(void)
-{
-	int err;
-	struct cloud_msg msg = {
-		.buf = ua_pattern,
-		.len = buttons_to_capture,
-		.endpoint = {
-			.type = CLOUD_EP_TOPIC_PAIR
-		}
-	};
-
-	atomic_set(&pattern_recording, 0);
-
-	err = cloud_send(cloud_backend, &msg);
-	if (err) {
-		printk("Could not send association message, error: %d\n", err);
-		cloud_error_handler(err);
+		/* if the association is not done soon enough (< ~5 min?)
+		 * a connection cycle is needed... TBD why.*/
```

Comment:
I asked Patrick and he was unaware of any sort of AWS timeout. 
Note that this behavior is *not* present on the LTE BLE gateway example.
So it may be something with the socket handling?